docs/permission/plugin-authors/authorize-the-read-endpoint.md

```diff
@@ -0,0 +1,133 @@
+---
+id: authorize-the-read-endpoint
+title: Authorize the read endpoint
+description: Authorize the read endpoint
+---
+
+Authorizing the `GET /todos` is similar to the update endpoint: whenever a `GET /todos` request is received, only the items that the user has created should be returned.
+
+As in the previous case, the permission policy can't take the decision itself, meaning that a conditional decision should be returned.
+Also, here we don't have a `resourceRef` but a list of resources.
+
+Potentially, something like this could be done:
+
+```diff
+    router.get('/todos', async (req, res) => {
++     const token = IdentityClient.getBearerToken(req.header('authorization'));
+
+-     res.json(getAll())
++     const items = getAll();
++     const decisions = await permissions.authorize(
++       items.map(({ id }) => ({ permission: todosListRead, resourceRef: id })),
++     );
+
++     const filteredItems = decisions.filter(
++       decision => decision.result === AuthorizeResult.ALLOW,
++     );
++     res.json(filteredItems);
+    });
+```
+
+This should do the trick. However, this approach has a downside. It would force us to retrieve all the elements upfront and authorize them one by one, and mean that concerns like pagination that are handled by the data source today also need to move into the plugin.
+
+To avoid this situation, the permissions framework has support for filtering items in the data source itself.
+
+Update `plugins/todo-list-backend/src/service/rules.ts`
+
+```diff
+  import { makeCreatePermissionRule } from '@backstage/plugin-permission-node';
+- import { Todo } from './todos';
++ import { Todo, TodoFilter } from './todos';
+
+  const createTodoListPermissionRule = makeCreatePermissionRule<
+    Todo,
+-   undefined
++   TodoFilter
+  >();
+
+  export const isOwner = createTodoListPermissionRule({
+    name: 'IS_OWNER',
+    description: 'Should allow only if the todo belongs to the user',
+    apply: (resource, userId) => {
+      return resource.author === userId;
+    },
+    toQuery: userId => {
+-     throw new Error('toQuery not implemented');
++     return resource => resource.author === userId;
```

Comment:
Perhaps it would be helpful to use some kind of declarative object to describe the filter here - I think the similarity between the `toQuery` and `apply` methods here has the potential to be a bit confusing. 

There's also an opportunity to mention somewhere in here the requirement that filters must be composable via anyOf/allOf/not.